ntbs-service/Pages/Shared/_SideMenu.cshtml

```diff
@@ -1,103 +1,26 @@
-                Previous History
-            </a>
-        </li>
-        <li class="@("contents-list-item " + (currentPage == NotificationSubPaths.EditTreatmentEvents ? "contents-list-item--current" : ""))">
-            <a class="app-subnav__link govuk-link govuk-link--no-visited-state" href="@RouteHelper.GetNotificationPath(Model.NotificationId, NotificationSubPaths.EditTreatmentEvents)">
-                Treatment Events
-            </a>
-        </li>
-        @if (Model.Notification.IsMdr)
+        @foreach (NotificationSection section in Enum.GetValues(typeof(NotificationSection)))
         {
-            <li class="@("contents-list-item " + (currentPage == NotificationSubPaths.EditMDRDetails ? "contents-list-item--current" : ""))">
-                <a class="app-subnav__link govuk-link govuk-link--no-visited-state" href="@RouteHelper.GetNotificationPath(Model.NotificationId, NotificationSubPaths.EditMDRDetails)">
-                    MDR Details
-                </a>
-            </li>
-        }
-        @if (Model.Notification.IsMBovis)
-        {
-            <li class="@("contents-list-item " + (currentPage == NotificationSubPaths.EditMBovisExposureToKnownCases ? "contents-list-item--current" : ""))">
-                <a class="app-subnav__link govuk-link govuk-link--no-visited-state" href="@RouteHelper.GetNotificationPath(Model.NotificationId, NotificationSubPaths.EditMBovisExposureToKnownCases)">
-                    M. bovis - Other cases
-                </a>
-            </li>
-            <li class="@("contents-list-item " + (currentPage == NotificationSubPaths.EditMBovisUnpasteurisedMilkConsumptions ? "contents-list-item--current" : ""))">
-                <a class="app-subnav__link govuk-link govuk-link--no-visited-state" href="@RouteHelper.GetNotificationPath(Model.NotificationId, NotificationSubPaths.EditMBovisUnpasteurisedMilkConsumptions)">
-                    M. bovis - Unpasteurised milk
-                </a>
-            </li>
-            <li class="@("contents-list-item " + (currentPage == NotificationSubPaths.EditMBovisOccupationExposures ? "contents-list-item--current" : ""))">
-                <a class="app-subnav__link govuk-link govuk-link--no-visited-state" href="@RouteHelper.GetNotificationPath(Model.NotificationId, NotificationSubPaths.EditMBovisOccupationExposures)">
-                    M. bovis - Occupation
-                </a>
-            </li>
-            <li class="@("contents-list-item " + (currentPage == NotificationSubPaths.EditMBovisAnimalExposures ? "contents-list-item--current" : ""))">
-                <a class="app-subnav__link govuk-link govuk-link--no-visited-state" href="@RouteHelper.GetNotificationPath(Model.NotificationId, NotificationSubPaths.EditMBovisAnimalExposures)">
-                    M. bovis - Animal exposure
-                </a>
-            </li>
+            if ((!section.IsMdr() || Model.Notification.IsMdr) 

```

Comment:
nice commonisation, but I don't think I get this conditions

Reply:
Gah, I started thinking about it and almost talked myself into thinking they were wrong :P But no, I think they're correct, and the logic is effectively describing that only when `sectionIsMdr _implies_ notifcationIsMdr` then we should shot that entry. This is how you can get there 

`(!A) || (A && B)` ~ `(!A || A) && (!A || B)` ~ `!A || B` ~ `A -> B`
This is saying only show the MDR section, if the notification needs one (as `!A || B` ~ `A -> B`)

Another way to describe it is (effectively the LHS of that derivation):
- in general, show all sections that are not mdr (`if (!section.IsMdr())`)...
- and for the mdr section, only show it when the notification is mdr